File: 3932-count-k-th-roots-in-a-range/3932-count-k-th-roots-in-a-range.kt
class Solution {
    fun countKthRoots(l: Int, r: Int, k: Int): Int {
        if (k == 1) {
            return r - l + 1
        }

        var result = 0
        var x = 0

        while (true) {
            var value = 1L

            repeat(k) {
                value *= x.toLong()
                if (value > r) {
                    return result
                }
            }

            if (value >= l) {
                result++
            }

            x++
        }
    }
}

File: 3932-count-k-th-roots-in-a-range/3932-count-k-th-roots-in-a-range.ts
function countKthRoots(l: number, r: number, k: number): number {
    if (k === 1) {
        return r - l + 1;
    }

    let result = 0;
    let x = 0;

    while (x ** k <= r) {
        if (x ** k >= l) {
            result++;
        }
        x++;
    }

    return result;
}